### PR TITLE
CAMEL-20115: Support for Start Date and End Date in camel-quartz

### PR DESCRIPTION
**Pull request by prasanthrao**
# Description

Fix for [CAMEL-20115](https://issues.apache.org/jira/browse/CAMEL-20115) Support for Start Date and End Date in camel-quartz

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL-20115) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by oscerd — COMMENTED** [review]
It looks good, but I think it should go on main, since it is a new feature.

File: components/camel-quartz/src/main/java/org/apache/camel/component/quartz/CamelJob.java [hunk 1]
```diff
@@ -94,6 +101,18 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
         }
     }
 
+    private boolean hasTriggerExpired(JobExecutionContext context) {
+        Date fireTime = context.getFireTime();
+        if (context.getTrigger().getStartTime() != null && fireTime.before(context.getTrigger().getStartTime())) {
+            // Trigger invalid as Fire Time is before Start Time
+            return true;
+        } else if (context.getTrigger().getEndTime() != null && fireTime.after(context.getTrigger().getEndTime())) {
```
Comment (davsclaus) [inline components/camel-quartz/src/main/java/org/apache/camel/component/quartz/CamelJob.java:109]: I think this should not be an else if, as you can have set BOTH start and end time, and then you need to check for both of them.

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: components/camel-quartz/src/main/java/org/apache/camel/component/quartz/CamelJob.java (hunk 1 above)
Comment (prasanthrao) [reply components/camel-quartz/src/main/java/org/apache/camel/component/quartz/CamelJob.java]: Indeed I missed the case, and corrected it in the new commit.

**Comment by oscerd** [comment]
Shouldn't this need to go on main first?

**Comment by prasanthrao** [comment]
> Shouldn't this need to go on main first?

Should I also create a PR for the main branch?

**Comment by davsclaus** [comment]
3.1.4 is EOL and we should only do this change in newer releases

**Pull request closed without merging**